worldedit-bukkit/src/main/java/com/fastasyncworldedit/bukkit/adapter/NMSAdapter.java

```diff
@@ -192,6 +199,9 @@ protected static void beginChunkPacketSend(String worldName, IntPair pair, Stamp
      * @since 2.12.0
      */
     protected static void endChunkPacketSend(String worldName, IntPair pair, StampLockHolder lockHolder) {
+        if (Fawe.isMainThread()) {
```

Comment:
In the current code, this always happens on the main thread or am I missing something?